### ISSUE TITLE
Restore notFound message key in API?

### ISSUE THREAD

**Issue by mikhailnov** [bigbluebutton/bigbluebutton#14210]
We are using a fork of the old 1.x version of the BigBlueButton WordPress plugin. Listing meeting rooms stopped working some time ago. Investigation showed that PHP code of the WordPress plugin expected to receive meesageKey == "notFound" in the API response, but received messageKey == "validationError".

I had to change php code in this way:
```diff
diff --git a/bigbluebutton-plugin.php b/bigbluebutton-plugin.php
index 5173226..cc592ce 100644
--- a/bigbluebutton-plugin.php
+++ b/bigbluebutton-plugin.php
@@ -1262,7 +1262,11 @@ function bigbluebutton_list_meetings() {
         if(!$info) {//If the server is unreachable, then prompts the user of the necessary action
             $out .= '<div class="updated"><p><strong>Unable to display the meetings. Please check the url of the bigbluebutton server AND check to see if the bigbluebutton server is running.</strong></p></div>';
             return $out;
-        } else if( $info['returncode'] == 'FAILED' && $info['messageKey'] != 'notFound' && $info['messageKey'] != 'invalidPassword') { /// If the meeting was unable to be deleted due to an error
+        } else if( $info['returncode'] == 'FAILED' &&
+                   $info['messageKey'] != 'notFound' &&
+                   $info['messageKey'] != 'validationError' &&
+                   $info['messageKey'] != 'invalidPassword'
+                 ) { /// If the meeting was unable to be deleted due to an error
             if($info['messageKey'] == 'checksumError') {
                 $out .= '<div class="updated"><p><strong>A checksum error occured. Make sure you entered the correct salt.</strong></p></div>';
             }
@@ -1270,7 +1274,12 @@ function bigbluebutton_list_meetings() {
                 $out .= '<div class="updated"><p><strong>'.$info['message'].'</strong></p></div>';
             }
             return $out;
-        } else if( $info['returncode'] == 'FAILED' && ($info['messageKey'] == 'notFound' || $info['messageKey'] != 'invalidPassword') ) { /// The meeting exists only in the wordpress db
+        // messageKey == notFound in older BigBlueButton versions, validationError in v2.4
+        } else if( $info['returncode'] == 'FAILED' &&
+                  ($info['messageKey'] == 'notFound' ||
+                  ($info['messagKey'] == 'validationError' && strpos($info['message'], 'meeting with that ID does not exist'))
+                  || $info['messageKey'] != 'invalidPassword')
+                 ) { /// The meeting exists only in the wordpress db
             if(!$printed) {
                 $out .= bigbluebutton_print_table_header();
                 $printed = true;
```

`$info['messageKey'] == 'notFound'` seems to be a better approach than `$info['messagKey'] == 'validationError' && strpos($info['message'], 'meeting with that ID does not exist'))`.

May be it is better to restore reporting notFound error?

From syslog of BigBlueButton:
```
run-prod.sh[5416]: 2022-01-26T11:37:22.986Z DEBUG o.b.web.controllers.ApiController - ApiController#getMeetingInfo
run-prod.sh[5416]: 2022-01-26T11:37:22.986Z INFO  o.b.api.service.ValidationService - Validating getMeetingInfo request with query string meetingID=yyy&password=password&checksum=xxx
run-prod.sh[5416]: 2022-01-26T11:37:22.986Z INFO  o.b.a.m.v.MeetingExistsValidator - Validating existence of meeting with ID yyy
run-prod.sh[5416]: 2022-01-26T11:37:22.986Z INFO  o.b.api.service.ServiceUtils - Attempting to find meeting with ID yyy
run-prod.sh[5416]: 2022-01-26T11:37:22.986Z INFO  o.b.api.service.ServiceUtils - Meeting with ID yyy could not be found
```

I am running not the newset version of BBB, sorry if it has already been fixed.
```
# dpkg -l | grep ^ii | grep bbb | awk '{print $2,$3}' | grep ^bbb
bbb-apps-akka 2.4.0-49
bbb-config 1:2.4.0-27
bbb-etherpad 1:2.4.0-2
bbb-freeswitch-core 2:2.4.0-3
bbb-freeswitch-sounds 1:1.6.7-1
bbb-fsesl-akka 2.4.0-16
bbb-html5 1:2.4.0-2323
bbb-learning-dashboard 1:2.4.0-8
bbb-libreoffice-docker 1:2.4.0-5
bbb-mkclean 1:0.8.7-1
bbb-playback 1:2.4.0-13
bbb-playback-presentation 1:2.4.0-7
bbb-record-core 1:2.4.0-9
bbb-web 1:2.4.0-43
bbb-webrtc-sfu 1:2.4.0-20
```

**Comment by mikhailnov** [comment]
Also, it gets validationError 'Provided moderator password is incorrect' when calling endMeeting on a non-existing meeting.
It seems to be something like https://github.com/bigbluebutton/bigbluebutton/issues/13220

**Comment by antobinary** [comment]
Hi @mikhailnov 
I wonder if this is resolved on BBB 2.4.6+, see https://github.com/bigbluebutton/bigbluebutton/pull/14668
